### PR TITLE
Wip/hughsie/unsigned payload

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -6829,9 +6829,9 @@ fu_engine_add_device(FuEngine *self, FuDevice *device)
 	    !fu_device_has_flag(device, FWUPD_DEVICE_FLAG_SIGNED_PAYLOAD) &&
 	    !fu_device_has_flag(device, FWUPD_DEVICE_FLAG_UNSIGNED_PAYLOAD) &&
 	    !fu_device_has_internal_flag(device, FU_DEVICE_INTERNAL_FLAG_MD_SET_SIGNED)) {
-		g_warning("%s device does not define payload %s [%s]",
+		g_warning("%s [%s] does not declare signed/unsigned payload -- perhaps add "
+			  "fu_device_add_flag(device, FWUPD_DEVICE_FLAG_UNSIGNED_PAYLOAD);",
```
Comment (superm1) [inline src/fu-engine.c:6833]: Who does the warning actually help?  I would argue it's not really useful to an end user.  Maybe:

```suggestion
		g_warning("%s [%s] does not declare if it accepts unsigned payloads",
```

Comment (hughsie) [reply src/fu-engine.c:6833]: I think we need to tell the programmer what do to.

Comment (superm1) [reply src/fu-engine.c:6833]: But an end user will see this.  Can we maybe only show that warning when supported build isn't set?

Comment (superm1) [reply src/fu-engine.c:6833]: Oh wait, this block is in that already.  Carry on :)

**Review by superm1 — APPROVED** [review]

**Pull request merged**